odoo/addons/base/models/ir_model.py

```diff
@@ -1718,26 +1718,28 @@ def check(self, model, mode='read', raise_exception=True):
 
         if not r and raise_exception:
             groups = '\n'.join('\t- %s' % g for g in self.group_names_with_access(model, mode))
-            msg_heads = {
+            raw_msg_heads = {
                 # Messages are declared in extenso so they are properly exported in translation terms
-                'read': _("Sorry, you are not allowed to access documents of type '%(document_kind)s' (%(document_model)s)."),
-                'write':  _("Sorry, you are not allowed to modify documents of type '%(document_kind)s' (%(document_model)s)."),
-                'create': _("Sorry, you are not allowed to create documents of type '%(document_kind)s' (%(document_model)s)."),
-                'unlink': _("Sorry, you are not allowed to delete documents of type '%(document_kind)s' (%(document_model)s)."),
-            }
-            msg_params = {
-                'document_kind': self.env['ir.model']._get(model).name or model,
-                'document_model': model,
+                'read': _("You are not allowed to access '%(document_kind)s' (%(document_model)s) records."),
+                'write':  _("You are not allowed to modify '%(document_kind)s' (%(document_model)s) records."),
+                'create': _("You are not allowed to create '%(document_kind)s' (%(document_model)s) records."),
+                'unlink': _("You are not allowed to delete '%(document_kind)s' (%(document_model)s) records."),
             }
+            msg_heads = _(
```

Comment:
This is not correct, and would have been detected by: https://github.com/odoo/odoo/pull/53683

Reply:
So, I should str() cast the parameters ?

Reply:
@sebgeelen see https://www.odoo.com/documentation/master/reference/translations.html#variables and https://www.odoo.com/documentation/master/reference/guidelines.html#use-translation-method-correctly for more examples. The first argument of the `_` method should always be a plain string, never a variable or an operation (this is what the pylint test added at #53683 will check)

Reply:
ok thanks, will check